controller/scheduler/k8s.py

```diff
@@ -0,0 +1,647 @@
+        for _ in xrange(30):
+            js_template = self._get_rc_(name)
+            if js_template["metadata"]["resourceVersion"] != resource_ver:
+                break
+            time.sleep(1)
+        self._get_schedule_status(name, num)
+        for _ in xrange(120):
+            count = 0
+            status, data, reason = self._get_pods()
+            parsed_json = json.loads(data)
+            for pod in parsed_json['items']:
+                if(pod['metadata']['generateName'] == name+'-' and
+                   pod['status']['phase'] == 'Running'):
+                    count += 1
+            if count == num:
+                break
+            time.sleep(1)
+
+    def _scale_app(self, name, num):
+        js_template = self._get_rc_(name)
+        js_template["spec"]["replicas"] = num
+        self._scale_rc(js_template)
+
+    def scale(self, name, image, command, **kwargs):
+        rc_name = name.replace(".", "-")
+        rc_name = rc_name.replace("_", "-")
+        if not 200 <= self._get_rc_status(rc_name) <= 299:
+            self.create(name, image, command, **kwargs)
+            return
+        name = name.replace(".", "-")
+        name = name.replace("_", "-")
+        num = kwargs.get('num', {})
+        js_template = self._get_rc_(name)
+        old_replicas = js_template["spec"]["replicas"]
+        try:
+            self._scale_app(name, num)
+        except Exception as e:
+            self._scale_app(name, old_replicas)
+            err = '{} (Scale): {}'.format(name, e)
+            raise RuntimeError(err)
+
+    def _create_rc(self, name, image, command, **kwargs):
+        container_fullname = name
+        app_name = kwargs.get('aname', {})
+        app_type = name.split(".")[1]
+        container_name = app_name+"-"+app_type
+        name = name.replace(".", "-")
+        name = name.replace("_", "-")
+        args = command.split()
+
+        num = kwargs.get('num', {})
+        l = {}
+        l["name"] = name
+        l["id"] = app_name
+        l["appversion"] = kwargs.get('version', {})
+        l["version"] = self.apiversion
+        l["image"] = self.registry+"/"+image
+        l['num'] = num
+        l['containername'] = container_name
+        l['type'] = app_type
+        template = string.Template(RC_TEMPLATE).substitute(l)
+        js_template = json.loads(template)
+        containers = js_template["spec"]["template"]["spec"]["containers"]
+        containers[0]['args'] = args
+        loc = locals().copy()
+        loc.update(re.match(MATCH, container_fullname).groupdict())
+        mem = kwargs.get('memory', {}).get(loc['c_type'])
+        cpu = kwargs.get('cpu', {}).get(loc['c_type'])
+        if mem or cpu:
+            containers[0]["resources"] = {"limits": {}}
+        if mem:
+            if mem[-2:-1].isalpha() and mem[-1].isalpha():
+                mem = mem[:-1]
+            mem = mem+"i"
+            containers[0]["resources"]["limits"]["memory"] = mem
```

Comment:
Currently this is how kubernetes scheduler works as far as i know.There is open issue(https://github.com/GoogleCloudPlatform/kubernetes/issues/168) where they want to take the resource request into consideration for scheduling than the limits.